### PR TITLE
fix: renamed DA flavor to standard and updated catalog.json

### PR DESCRIPTION
**Pull request by imprateeksh**
### Description
This PR addresses Issues: 
 - [Issue-205](https://github.com/terraform-ibm-modules/terraform-ibm-icd-elasticsearch/issues/205) - This is to provide a supporting document to explain the complex object types used as part of DA.
 - [Issue-206](https://github.com/terraform-ibm-modules/terraform-ibm-icd-elasticsearch/issues/206) - To remove the default elastic version and update `catalog.json`
 - [Issue-207](https://github.com/terraform-ibm-modules/terraform-ibm-icd-elasticsearch/issues/207) - To rename the `secure` flavor to `standard`

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content
Release Notes:

1. Removed default Elasticsearch version and updated `catalog.json` to include the supported elastic versions.
2. Renamed DA` flavor to `standard`.
3. Added supporting documentation for complex object types in DA(Deployable Architectures).

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [x] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by imprateeksh** [comment]
Upgrade test failure is seen - [TestRunStandardUpgradeTest](https://github.com/terraform-ibm-modules/terraform-ibm-icd-elasticsearch/actions/runs/9695898732/job/26756794053#step:7:176)

File: ibm_catalog.json [hunk 1]
```diff
@@ -170,7 +170,27 @@
                   ]
                 },
                 {
-                  "key": "elasticsearch_version"
+                  "key": "elasticsearch_version",
+                  "required": true,
+                  "default_value": "",
```
Comment (ocofaigh) [inline ibm_catalog.json:175]: don't add any `default_value` - let it default the the terraform default value

File: ibm_catalog.json [hunk 2]
```diff
@@ -170,7 +170,27 @@
...
+                  "options": [
+                    {
+                      "displayname": "v8.12 (preferred)",
```
Comment (ocofaigh) [inline ibm_catalog.json:178]: I would not add `preferred` here - as this may become out of date

File: ibm_catalog.json [hunk 3]
```diff
@@ -170,7 +170,27 @@
...
+                    {
+                      "displayname": "v8.10",
+                      "value": "8.10"
+                    },
+                    {
+                      "displayname": "v8.7",
+                      "value": "8.7"
+                    },
+                    {
+                      "displayname": "v7.17 (deprecated)",
```
Comment (ocofaigh) [inline ibm_catalog.json:190]: we should not have support for a deprecated version

File: solutions/standard/variables.tf [hunk 4]
```diff
@@ -41,7 +41,6 @@ variable "plan" {
 variable "elasticsearch_version" {
   description = "The version of the Elasticsearch instance. If no value is passed, the current preferred version of IBM Cloud Databases is used."
   type        = string
```
Comment (ocofaigh) [inline solutions/standard/variables.tf:43]: default value should be `null`

**Review by ocofaigh — CHANGES_REQUESTED** [review]
some comments

File: ibm_catalog.json (hunk 1 above)
Comment (imprateeksh) [reply ibm_catalog.json:175]: done.

File: ibm_catalog.json (hunk 2 above)
Comment (imprateeksh) [reply ibm_catalog.json:178]: sure, done

File: ibm_catalog.json (hunk 3 above)
Comment (imprateeksh) [reply ibm_catalog.json:190]: sure, will remove it.

File: solutions/standard/variables.tf (hunk 4 above)
Comment (imprateeksh) [reply solutions/standard/variables.tf:43]: Oh yes, this was a miss. Thank you.

File: ibm_catalog.json [hunk 5]
```diff
@@ -170,7 +170,22 @@
...
+                    {
+                      "displayname": "v8.12",
+                      "value": "8.12"
+                    },
+                    {
+                      "displayname": "v8.10",
+                      "value": "8.10"
+                    },
+                    {
+                      "displayname": "v8.7",
```
Comment (ocofaigh) [inline ibm_catalog.json:185]: our module doesn't even support this version? https://github.com/terraform-ibm-modules/terraform-ibm-icd-elasticsearch/blob/b392f929de1d1cea38d1007455e138090f996ac3/variables.tf#L51-L52

I suggest to remove the `v` from the display names too - its not something ICD adds, so why would we add it

File: solutions/standard/DA-types.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+The `service_credential_names` variable is a map where the key is the name of the service credential and the value is the role assigned to that credential.
+
+**Default Value:**
+The default value for the `service_credential_names` variable is an empty map (`{}`).
+
+### Detailed Structure:
+
+- `Key (name, required):` The name of the service credential.
+
+- `Value (role, required):` The role assigned to the service credential.
```
Comment (ocofaigh) [inline solutions/standard/DA-types.md:20]: suggest to give an example of what a typical value could look like

File: solutions/standard/DA-types.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+
+ - `role (optional):` Defines the role assigned to the user, determining their access level and permissions.
+
+
+**Default Value:**
+The default value for the `users` variable is an empty list (`[]`).
+
+**Sensitive:**
+The `users` variable is marked as sensitive due to the inclusion of passwords.
+
```
Comment (ocofaigh) [inline solutions/standard/DA-types.md:45]: suggest to give an example of what a typical value could look like

File: solutions/standard/DA-types.md [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+      io_over_period           = "10m"
+      rate_increase_percent    = 15
+      rate_limit_mb_per_member = 200000
+      rate_period_seconds      = 600
+      rate_units               = "mb"
+    })
+  })
+  description = "The rules to allow the database to increase resources in response to usage. Only a single autoscaling block is allowed. Make sure you understand the effects of autoscaling, especially for production environments. See https://cloud.ibm.com/docs/databases-for-elasticsearch?topic=databases-for-elasticsearch-autoscaling&interface=cli#autoscaling-considerations in the IBM Cloud Docs."
+  default     = null
+}
```
Comment (ocofaigh) [inline solutions/standard/DA-types.md:119]: this means nothing in Projects land. The example needs to show how a consumer would enter a value in Projects UI

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see comments

File: ibm_catalog.json (hunk 5 above)
Comment (imprateeksh) [reply ibm_catalog.json:185]: done.

File: solutions/standard/DA-types.md (hunk 8 above)
Comment (imprateeksh) [reply solutions/standard/DA-types.md:119]: Got it. Will provide only the values required.

File: solutions/standard/DA-types.md (hunk 6 above)
Comment (imprateeksh) [reply solutions/standard/DA-types.md:20]: Done.

Comment (imprateeksh) [reply solutions/standard/DA-types.md:19]: Provided example.

File: solutions/standard/DA-types.md (hunk 7 above)
Comment (imprateeksh) [reply solutions/standard/DA-types.md:56]: Done

File: solutions/standard/DA-types.md [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+**Default Value:**
+The default value for the `service_credential_names` variable is an empty map (`{}`).
+
+- `Key (name, required):` The name of the service credential.
+
+- `Value (role, required):` The role assigned to the service credential.
+
+**Sample Examples:**
+
+```json
```
Comment (ocofaigh) [inline solutions/standard/DA-types.md:23]: thats not json

File: solutions/standard/DA-types.md [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+  {
+    "elastic_user_1" = "Viewer"
+  }
+```
+```json
```
Comment (ocofaigh) [inline solutions/standard/DA-types.md:28]: thats not json

File: solutions/standard/DA-types.md [hunk 11]
```diff
@@ -0,0 +1,145 @@
...
+
+
+**Default Value:**
+The default value for the `users` variable is an empty list (`[]`).
+
+**Sensitive:**
+The `users` variable is marked as sensitive due to the inclusion of passwords.
+
+**Sample Example:** The below example shows the list of two users where one of the users is not provided the optional `role` value.
+```json
```
Comment (ocofaigh) [inline solutions/standard/DA-types.md:58]: thats not json

File: solutions/standard/DA-types.md [hunk 12]
```diff
@@ -0,0 +1,145 @@
...
+[
+    {
+      name     = "es_admin"
+      password = "securepassword123" # pragma: allowlist secret
```
Comment (ocofaigh) [inline solutions/standard/DA-types.md:62]: remove `# pragma: allowlist secret` and instead update the secrets baseline:
```
detect-secrets scan --update .secrets.baseline
detect-secrets audit .secrets.baseline
```

File: solutions/standard/DA-types.md [hunk 13]
```diff
@@ -0,0 +1,145 @@
...
+      type     = "admin"
+      role     = "admin"
+    },
+    {
+      name     = "es_reader"
+      password = "readpassword123" # pragma: allowlist secret
```
Comment (ocofaigh) [inline solutions/standard/DA-types.md:68]: remove `# pragma: allowlist secret` and instead update the secrets baseline:
```
detect-secrets scan --update .secrets.baseline
detect-secrets audit .secrets.baseline
```

File: solutions/standard/DA-types.md [hunk 14]
```diff
@@ -0,0 +1,145 @@
...
+      type     = "reader"
+    }
+]
+```
```
Comment (ocofaigh) [inline solutions/standard/DA-types.md:72]: Trying that value in projects fails with this:
image

File: solutions/standard/DA-types.md [hunk 15]
```diff
@@ -0,0 +1,145 @@
...
+```json
+  {
+    "elastic_user_1" = "Viewer"
+  }
+```
+```json
+  {
+    "elastic_user_2" = "Editor"
+  }
+```
```
Comment (ocofaigh) [inline solutions/standard/DA-types.md:32]: can you just give 1 map that has two roles in it?

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see latest comments

File: solutions/standard/DA-types.md [hunk 16]
```diff
@@ -0,0 +1,145 @@
...
+      "io_above_percent": 80,
+      "io_enabled": true,
+      "io_over_period": "10m",
+      "rate_increase_percent": 15,
+      "rate_limit_mb_per_member": 200000,
+      "rate_period_seconds": 600,
+      "rate_units": "mb"
+    }
+  }
+}
```
Comment (ocofaigh) [inline solutions/standard/DA-types.md:144]: This also did not work when I tried it:
```
 2024/06/28 10:27:55 Terraform plan | Error: Invalid value for input variable
 2024/06/28 10:27:55 Terraform plan | 
 2024/06/28 10:27:55 Terraform plan |   on schematics.tfvars line 27:
 2024/06/28 10:27:55 Terraform plan |   27: auto_scaling = {"type":{"disk":{"capacity_enabled":true,"free_space_less_than_percent":15,"io_above_percent":85,"io_enabled":true,"io_over_period":"10m","rate_increase_percent":20,"rate_limit_mb_per_member":5000000,"rate_period_seconds":600,"rate_units":"mb"},"memory":{"io_above_percent":80,"io_enabled":true,"io_over_period":"10m","rate_increase_percent":15,"rate_limit_mb_per_member":200000,"rate_period_seconds":600,"rate_units":"mb"}}}
 2024/06/28 10:27:55 Terraform plan | 
 2024/06/28 10:27:55 Terraform plan | The given value is not suitable for var.auto_scaling declared at
 2024/06/28 10:27:55 Terraform plan | variables.tf:160,1-24: attributes "disk" and "memory" are required.
```

Comment (imprateeksh) [reply solutions/standard/DA-types.md:144]: Oh!  I am looking into this.

File: solutions/standard/DA-types.md (hunk 13 above)
Comment (imprateeksh) [reply solutions/standard/DA-types.md:68]: done

File: solutions/standard/DA-types.md (hunk 12 above)
Comment (imprateeksh) [reply solutions/standard/DA-types.md:62]: done

File: solutions/standard/DA-types.md (hunk 9 above)
Comment (imprateeksh) [reply solutions/standard/DA-types.md:23]: changed to `hcl`

File: solutions/standard/DA-types.md (hunk 10 above)
Comment (imprateeksh) [reply solutions/standard/DA-types.md:28]: changed to `hcl`

File: solutions/standard/DA-types.md (hunk 11 above)
Comment (imprateeksh) [reply solutions/standard/DA-types.md:58]: changed to `hcl`

**Review by ocofaigh — CHANGES_REQUESTED** [review]
@imprateeksh can you remove the doc from this PR so we can merge the remaining code? And add the doc in a subsequent PR. We can publish the DA to community registry without the doc for now, but need the other changes

File: solutions/standard/DA-types.md (hunk 15 above)
Comment (imprateeksh) [reply solutions/standard/DA-types.md:32]: done

File: solutions/standard/DA-types.md (hunk 16 above)
Comment (imprateeksh) [reply solutions/standard/DA-types.md:144]: I am following this as per the [usage](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/database#sample-database-instance-by-using-auto_scaling) here. `mb` is a valid value here.

**Comment by imprateeksh** [comment]
> @imprateeksh can you remove the doc from this PR so we can merge the remaining code? And add the doc in a subsequent PR. We can publish the DA to community registry without the doc for now, but need the other changes

Yes sure @ocofaigh. Will do it right away.

**Comment by imprateeksh** [comment]
> > @imprateeksh can you remove the doc from this PR so we can merge the remaining code? And add the doc in a subsequent PR. We can publish the DA to community registry without the doc for now, but need the other changes
> 
> Yes sure @ocofaigh. Will do it right away.

@ocofaigh I have removed the doc.

**Review by ocofaigh — APPROVED** [review]

**Comment by ocofaigh** [comment]
@imprateeksh can you resolve conflicts?

**Comment by imprateeksh** [comment]
> @imprateeksh can you resolve conflicts?

Resolved

**Review by ocofaigh — APPROVED** [review]

File: solutions/standard/DA-types.md (hunk 14 above)
Comment (imprateeksh) [reply solutions/standard/DA-types.md]: This is getting saved without any error. However I noticed, it does not allows to save if the values are copied and pasted. On manually typing the values, these were saved without showing any error.
image

image

**Pull request merged**